File: progressbar/progressbar.cpp
#include <iostream>
#include <string>

#define len     50
using namespace std;

static char fWheel(int digit) {
    char wheel[5] = "\\|/-";

    return wheel[digit%4];
}

static string fSpaces(int digit, int tour) {
    string spaces;
    spaces.append("\r[");
    spaces.append(digit, '=');
    spaces.append(tour - digit - 3, ' ');
    spaces.append("]");

    return spaces;
}

static void fProgress(int digit) {
    for(digit = 1; digit < len - 2; digit++) {
        cout << fSpaces(digit, len) << '\t' << fWheel(digit) << ' ';

        usleep(40000);
        fflush(stdout);
    }
}

int main(void) {
    fProgress(len);

    cout << "\b\bDone.\n";
}

File: progressbar/progressbar.py
#!/usr/bin/python
#-*- coding: utf-8 -*-

import time
import sys

class Progress():
    def fWheel(self, wheel, digit):
        return wheel[digit%4]

    def fSpaces(self, tour, digit):
        return "[" + digit * "=" + (tour - digit - 1) * " " + "]"

    def fProgressbar(self, wheel, tour, digit):
        sys.stdout.write("\r%s\t%s " % (self.fSpaces(tour, digit),
                         self.fWheel(wheel, digit)))
        sys.stdout.flush()

    def __init__(self):
        wheel = ("\\", "|", "/", "-")
        tour = 50 - 2

        for digit in range(tour):
            self.fProgressbar(wheel, tour, digit)
            time.sleep(0.04)

        print("\b\bDone.")

if __name__ == '__main__':
    Progress()
